libass/c/c_shift_bitmap.c

```diff
@@ -30,29 +30,34 @@
  * expressed in 26.6 fixed point.
  * Pure C implementation.
  */
-void ass_shift_bitmap_c(uint8_t *restrict buf, ptrdiff_t s,
-                        size_t w, size_t h,
+void ass_shift_bitmap_c(uint8_t *restrict buf, ptrdiff_t stride,
+                        size_t width, size_t height,
                         uint32_t shift_x, uint32_t shift_y, uint16_t *restrict tmp)
 {
-    assert((shift_x & ~63) == 0 && (shift_y & ~63) == 0);
-
-    // Shift in x direction
-    if (shift_x)
-        for (int32_t y = 0; y < h; y++) {
-            for (int32_t x = w - 1; x > 0; x--) {
-                uint8_t b = buf[x + y * s - 1] * shift_x >> 6;
-                buf[x + y * s - 1] -= b;
-                buf[x + y * s] += b;
-            }
-        }
+    ASSUME(shift_x < 64 && shift_y < 64 && !(stride % 16) && width > 0 && height > 0 && stride > 0);
+
+    for (size_t y = 0; y < height; y++) {
+        uint16_t shifted_from_top = 0;
+        for (size_t x = 0; x < width; x++) {
+            uint16_t px = buf[x] << 1;
+            uint16_t shifted_from_left = tmp[x];
```

Comment:
`shifted_from__(top|left)` names are switched

Reply:
Whoops, must've happened during one of the find/replaces (I went through a few options). Fixed.